### PR TITLE
add storage account config to function apps

### PR DESCRIPTION
**Pull request by Benedikt-Fuchs-Crayon**
adds support for storage_account blocks in azurerm_<os>_function_app as described in https://github.com/hashicorp/terraform-provider-azurerm/issues/6790
also fixing a small copy paste mistake, where a error message falsely claims the wrong os

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @Benedikt-Fuchs-Crayon  - looks like we have some test failures to fix thou:

image

File: internal/services/appservice/windows_function_app_resource_test.go [hunk 1]
```diff
@@ -2836,3 +2887,149 @@ resource "azurerm_service_plan" "test" {
...
+
+  site_config {}
+
+  storage_account {
+    name         = "files"
+    type         = "AzureFiles"
+    account_name = azurerm_storage_account.test.name
+    share_name   = azurerm_storage_share.test.name
+    access_key   = azurerm_storage_account.test.primary_access_key
+    mount_path   = "/files"
```
Comment (jackofallops) [inline internal/services/appservice/windows_function_app_resource_test.go:2914]: This needs to be a path below `mounts` for Windows functions:

```
Details=[{"Message":"The parameter 'AzureStoragePropertyDictionary' has an invalid value.

...

MountPath: /files. MountPath must be sub-directory of \\mounts."}
```

File: internal/services/appservice/windows_function_app_resource_test.go [hunk 2]
```diff
@@ -2836,3 +2887,149 @@ resource "azurerm_service_plan" "test" {
...
+
+  site_config {}
+
+  storage_account {
+    name         = "updatedfiles"
+    type         = "AzureBlob"
+    account_name = azurerm_storage_account.test.name
+    share_name   = azurerm_storage_share.test.name
+    access_key   = azurerm_storage_account.test.primary_access_key
+    mount_path   = "/blob"
```
Comment (jackofallops) [inline internal/services/appservice/windows_function_app_resource_test.go:2945]: As above.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @Benedikt-Fuchs-Crayon 
Thanks for the changes, there's just a couple test config changes needed and I think this is good to go. 

Thanks!

**Comment by Benedikt-Fuchs-Crayon** [comment]
thank you for the feedback,
I changed the paths to `\\mounts\\files` and `\\mounts/blob` respectively

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @Benedikt-Fuchs-Crayon - still seeing some test failures related to this pr:

image

**Comment by jackofallops** [comment]
Hi @Benedikt-Fuchs-Crayon - To resolve the conflicts and extend the feature to slots I've pulled your changes onto a new branch/PR (I was unable to safely push to your fork as the PR originated from your `main` branch, sorry 🙈 ). I'm going to close this PR in favour of the new one, where I've kept as much as I can of your code while resolving the conflicts and extending the functionality to slots.

Thanks again for the contribution, and apologies for the delays in getting back to this!

**Pull request closed without merging**